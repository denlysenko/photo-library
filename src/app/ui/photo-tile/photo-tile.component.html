<div class="photo-tile card">
  <img class="photo"
       [src]="photo?.link"
       alt="photo">
</div>