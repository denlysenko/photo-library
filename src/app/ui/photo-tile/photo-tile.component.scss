.photo-tile {
  width: 280px;
  height: 280px;
  cursor: pointer;

  .photo {
    width: 100%;
    height: 100%;
  }
}
