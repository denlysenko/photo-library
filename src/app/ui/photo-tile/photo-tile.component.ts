import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Photo } from '../../interfaces';

@Component({
  selector: 'app-photo-tile',
  templateUrl: './photo-tile.component.html',
  styleUrls: ['./photo-tile.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PhotoTileComponent {
  @Input() photo: Photo;
  @Input() size: 'sm' | 'lg' = 'sm';
}
