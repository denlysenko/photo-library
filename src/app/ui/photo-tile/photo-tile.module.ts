import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { PhotoTileComponent } from './photo-tile.component';

@NgModule({
  imports: [CommonModule],
  declarations: [PhotoTileComponent],
  exports: [PhotoTileComponent],
})
export class PhotoTileModule {}
