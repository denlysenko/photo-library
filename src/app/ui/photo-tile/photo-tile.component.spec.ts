import { CommonModule } from '@angular/common';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PhotoTileComponent } from './photo-tile.component';

const photo = {
  id: 'photoId',
  link: 'link',
};

describe('PhotoTileComponent', () => {
  let component: PhotoTileComponent;
  let fixture: ComponentFixture<PhotoTileComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [CommonModule],
      declarations: [PhotoTileComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PhotoTileComponent);
    component = fixture.componentInstance;
    component.photo = photo;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });
});
