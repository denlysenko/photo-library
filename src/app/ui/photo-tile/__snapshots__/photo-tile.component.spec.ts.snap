// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoTileComponent should create 1`] = `
<app-photo-tile
  photo={[Function Object]}
>
  <div
    class="photo-tile card"
  >
    <img
      alt="photo"
      class="photo"
      src="link"
    />
  </div>
</app-photo-tile>
`;
